app/components/Editor.tsx

```diff
@@ -54,10 +54,24 @@ export type Props = Optional<
 };
 
 function Editor(props: Props, ref: React.RefObject<SharedEditor> | null) {
-  const { id, shareId, onChange, onHeadingsChange } = props;
+  const { id, shareId, onChange, onHeadingsChange, embedIntegrations } = props;
   const { documents } = useStores();
   const { showToast } = useToasts();
   const dictionary = useDictionary();
+  const embedsWithSetings = React.useMemo(
```

Comment:
I'd consider pulling this into a local hook, `useEmbedIntegrations` or similar, we can pull the info from the store in the hook and totally avoid passing `embedIntegrations` into this component.

Reply:
> we can pull the info from the store in the hook

_Pull from the server from within the hook_? As it's currently done in [`DataLoader.tsx`](https://github.com/outline/outline/pull/3980/files#diff-18ef7e911d2802727b4354c37055ae38684c757f7de708d9010180f2ea98d248R82)? Or keep the API call in `DataLoader.tsx` and just do `return integrations.orderedData(...)` in the hook? The latter should work too since `DataLoader.tsx` mounts before `Editor.tsx`.

Reply:
I was thinking about that, we could co-locate the data loading yes. Later we may want to support these integrations on public share links, that _may_ mean moving the data loading again. But it would be neat to include it in the hook for now

Reply:
Fixed!